addons/web/static/tests/views/fields/state_selection_field_tests.js

```diff
@@ -508,4 +508,40 @@ QUnit.module("Fields", (hooks) => {
         );
         assert.deepEqual(dropdownItemTexts, ["Custom blocked", "Custom done"]);
     });
+
+    QUnit.test("works when required in a readonly view ", async function (assert) {
+        serverData.models.partner.records[0].selection = "normal";
+        serverData.models.partner.records = [serverData.models.partner.records[0]];
+        const view = await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <kanban>
+                    <templates>
+                        <t t-name="kanban-box">
+                            <div>
+                                <field name="selection" widget="state_selection" required="1"/>
+                            </div>
+                        </t>
+                    </templates>
+                </kanban>`,
+            mockRPC: (route, args, performRPC) => {
+                if (route === "/web/dataset/call_kw/partner/write") {
+                    assert.step("write");
+                }
+                return performRPC(route, args);
+            },
+        });
+
+        await click(target, ".o_field_state_selection button");
+        const doneItem = target.querySelectorAll(".dropdown-item")[1]; // item "done";
+        await click(doneItem);
+
+        await view.model.load();
+        await nextTick();
```

Comment:
why do you do that?

Reply:
@aab-odoo By reloading the model, if the new value hasn't be stored, it gets back to the wrong state. 
The next tick seemed necessary for the ui to show the wrong state color.
I do a double check in this test basically: I check for the write in mockRPC, and I check the UI for the good color.

Reply:
I don't like doing that kind of manual, technical operations in tests. It's white box and has an higher maintenance cost than a fully blackbox test. So here I'd either:
  - do not do the reload and the further assertion as it looks redundant anyway (maybe assert that we write the correct value though)
  - produce the reloading by an actual flow (probably the one that you did outside the test environment to check that your dev was correct), for instance click edit/save

Reply:
@aab-odoo I've removed the "useless" double check.